vm/systemSmartContracts/stakingWaitingList.go

```diff
@@ -801,6 +801,60 @@ func (s *stakingSC) stakeNodesFromQueue(args *vmcommon.ContractCallInput) vmcomm
 	return vmcommon.Ok
 }
 
+func (s *stakingSC) unStakeAllNodesFromQueue(args *vmcommon.ContractCallInput) vmcommon.ReturnCode {
+	if !s.enableEpochsHandler.IsFlagEnabled(common.StakingV4Step1Flag) {
+		s.eei.AddReturnMessage("invalid method to call")
+		return vmcommon.UserError
+	}
+	if !bytes.Equal(args.CallerAddr, s.endOfEpochAccessAddr) {
+		s.eei.AddReturnMessage("stake nodes from waiting list can be called by endOfEpochAccess address only")
+		return vmcommon.UserError
+	}
+	if len(args.Arguments) != 0 {
+		s.eei.AddReturnMessage("number of arguments must be equal to 0")
+		return vmcommon.UserError
+	}
+
+	waitingListData, err := s.getFirstElementsFromWaitingList(math.MaxUint32)
+	if err != nil {
+		s.eei.AddReturnMessage(err.Error())
+		return vmcommon.UserError
+	}
+	if len(waitingListData.blsKeys) == 0 {
+		s.eei.AddReturnMessage("no nodes in queue")
+		return vmcommon.Ok
+	}
+
+	nodePriceToUse := big.NewInt(0).Set(s.minNodePrice)
+	if s.enableEpochsHandler.IsFlagEnabled(common.CorrectLastUnJailedFlag) {
+		nodePriceToUse.Set(s.stakeValue)
+	}
+
+	for i, blsKey := range waitingListData.blsKeys {
+		registrationData := waitingListData.stakedDataList[i]
+
+		registrationData.Staked = false
```

Comment:
should we reuse the function tryUnStake for L836-L845?

Reply:
we cannot use that one, as that is for staked nodes, not nodes in queue.